components/tooltip/Tooltip.js

```diff
@@ -250,22 +213,33 @@ const tooltipFactory = (options = {}) => {
           ComposedComponent,
           finalProps,
           children,
-          visible && (
-            <Portal>
-              <div
-                ref={node => {
-                  this.tooltipNode = node;
-                }}
-                className={classNames}
-                data-teamleader-ui="tooltip"
-                style={{ top, left }}
-              >
-                <Box className={theme['inner']} {...SIZES[tooltipSize]}>
-                  {tooltipIcon && <div className={theme['icon']}>{tooltipIcon}</div>}
-                  <div className={theme['text']}>{tooltip}</div>
-                </Box>
-              </div>
-            </Portal>
+          createPortal(
+            <Transition
+              in={active}
+              onExited={() => {
+                document.body.removeChild(this.tooltipRoot);
```

Comment:
Handling DOM operations in `render()` doesn't feel like the best place to do so.

Reply:
I don't like it either, but it's the easiest way to do I guess. I think you can't use `componentWillUnmount` because it will remove the element before the transition happens, not sure though. Did you try this @driesd?

And also extract the function out of the render method.

Reply:
I extracted the arrow function out of render method, but I'm not able to use the `componentWillUnmount` lifecycle method instead. This is because it's fired to early and so it breaks the exit animation.

 :ok: by @driesd on 27/11/2017 14:23:56